### PR TITLE
Typescriptify GroupingChooser

### PR DESCRIPTION
**Pull request by TomTirapani**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [ ] Caught up with `develop` branch as of last change.
- [ ] Added CHANGELOG entry, or determined not required.
- [ ] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [ ] Updated doc comments / prop-types, or determined not required.
- [ ] Reviewed and tested on Mobile, or determined not required.
- [ ] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/grouping/GroupingChooser.ts [hunk 1]
```diff
@@ -51,7 +50,7 @@ export const [GroupingChooser, groupingChooser] = hoistCmp.withFactory({
             ...layoutProps,
             item: popover({
                 isOpen,
-                popoverRef: (v) => model.popoverRef(v), // Workaround for #2272
+                popoverRef: (v) => (model.popoverRef as any)(v), // Workaround for #2272
```
Comment (TomTirapani) [inline desktop/cmp/grouping/GroupingChooser.ts:53]: Not sure if this is the best way to do this, but Typescript wasn't happy with calling methods on the ref

File: desktop/cmp/button/Button.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ export interface ButtonProps extends BoxProps {
     icon?: ReactElement;
     intent?: Intent,
     minimal?: boolean,
-    onClick?: () => void,
+    onClick?: (e: any) => void,
```
Comment (lbwexler) [inline desktop/cmp/button/Button.ts:101]: just caught this as well, but added it as type :MouseEvent.  Will merge

File: desktop/cmp/grouping/GroupingChooser.ts (hunk 1 above)
Comment (lbwexler) [reply desktop/cmp/grouping/GroupingChooser.ts:53]: I can look at the type we are passing back for createObservableRef now.  Maybe it needs to be improved.  Note that it is generic, so you can say indicate what type your Ref is going to eventually refer to/receive.

File: desktop/cmp/grouping/GroupingChooser.ts [hunk 3]
```diff
@@ -88,41 +87,33 @@ export const [GroupingChooser, groupingChooser] = hoistCmp.withFactory({
...
-    ...Button.propTypes,
-
+export interface GroupingChooserProps extends ButtonProps {
     /** Text to represent empty state (i.e. value = null or []) */
-    emptyText: PT.string,
+    emptyText?: string,
 
     /** Primary component model instance. */
-    model: PT.instanceOf(GroupingChooserModel),
+    model?: GroupingChooserModel,
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply desktop/cmp/grouping/GroupingChooser.ts:95]: its a good question --- we could go either way!  I think yes, better to pass it through as a general convention.  type could default to null or void, to indicate that most buttons don't actually have a model

**Comment by lbwexler** [comment]
good catch, fixed.

**Pull request merged**